src/CommandingOfficers/Commander.java

```diff
@@ -254,10 +178,20 @@ public boolean isEnemy(Commander other)
     return true;
   }
 
-  // Leaving this for now, to avoid merge conflicts
-  public UnitModel getUnitModel(UnitEnum unitType)
+  public UnitModel getUnitModel(UnitRoleEnum unitType)
   {
-    return unitModels.get(unitType);
+    UnitModel um = null;
+
+    for( UnitModel iter : unitModels )
+    {
+      if( iter.role == unitType )
```

Comment:
Since this just returns the first unit with the matching role, we can't get specific units anymore. Hrm.

Reply:
That's sort of intentional, since I'd like the rest of the game to work regardless of unit set, so long as the unit set follows our assumptions.

Reply:
Hm. What happens when we make campaign missions where three battleships appear 10 turns into the match? Or do you just want to cross that bridge when we get to it?

Reply:
We do have unit-fetching based on string lookup right now, which is what we use to deal with map init. This doesn't necessarily need to be different.

I will say that such missions would likely only be balanced with one unit set in mind, since trying to set up both DoR and standard Battleships (and my design, when that happens) to be the same strength in the same place at the same time would be... interesting.

If the missions are defined as external/text files, it's going to be exactly the same as map import.
If the missions are defined in code, we have even more flexibility.